pages/architecture/distributed_registry.rst

```diff
@@ -0,0 +1,230 @@
+
+======================  ===========  ====
+Field                   Format       Description
+======================  ===========  ====
+ID                      UUID / hash  Uniquely identifies this event
+Previous event ID       UUID / hash  Refers to the previous event in this stream
+Date of issue           timestamp    Timestamp the event was emitted
+Signature               TBD          Signature which authenticates the event
+======================  ===========  ====
+
+Processes
+*********
+
+Registering a vendor
+--------------------
+This process makes a vendor known to the network. Afterwards:
+
+* The vendor is known to the network.
+* The vendor van register their contracts with care organisations.
+
+Limitations:
+
+* For now, we only support 1 node per vendor.
+* For now, we only support 1 vendor per care organisation (but a vendor can serve multiple care organisations).
+
+Steps:
+
+#. Node registration:
+    #. The vendor generates a CSR (for the node's CA certificate) and submits it to Nuts Discovery.
+    #. A Nuts administrator accepts the CSR and issues the node CA certificate.
+    #. The vendor issues node identity certificate (for its node) using its CA.
+    #. The vendor submits its node information (node identity certificate, node address, ...) to Nuts Discovery.
+    #. The node information is published so that every (other) node in the network knows about the vendor's node.
+#. Vendor registration:
+    #. The vendor generates a CSR (for the vendor's CA certificate) and submits it to Nuts Registry.
+       TODO: Nuts Registry shouldn't be responsible for issuing certificates, since we're also issuing them
+       from Nuts Discovery? Or maybe they're RA's in front of the Nuts Foundation CA?
+    #. A Nuts administrator accepts the CSR and issues the vendor CA certificate.
+    #. The vendor creates the registration event and signs it using the issued CA certificate.
+
+VendorRegistration Event
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+======================  =====
+Field                   Format
+======================  =====
+Vendor CA certificate   PEM encoded X.509
+======================  =====
+
+Validation
+""""""""""
+
+#. Is the vendor CA certificate signed by the Nuts Foundation?
+
+
+Registering a contract (vendor - care organisation)
+---------------------------------------------------
+
+Through this process a vendor registers a care organisation as its client.
+Afterwards, the vendor can register endpoints for the care organisation it serves through its node.
+
+Steps:
+
+#. The vendor generates a CSR for the vendor's contract with the care organisation and signs it
+   using its vendor CA certificate (see the table below for the prescribed fields of the certificate).
+#. The vendor creates the contract proposal and signs it using the (just issued) contract certificate.
+#. The vendor creates the contract approval and signs it (again) using the contract certificate (see note below).
```

Comment:
Not needed if this flow is modelled as multiple events. Where the current business logic just requires a single sig.